pkg/node/manager/manager.go

```diff
@@ -827,6 +829,13 @@ func (m *manager) NodeDeleted(n nodeTypes.Node) {
 	}
 }
 
+// NodeSync signals the manager that the initial nodes listing (either from k8s or kvstore)
+// has been completed. This allows the manager to initiate the deletion of possible stale entries
+// in the kernel IP sets managed by Cilium.
+func (m *manager) NodeSync() {
+	m.ipsetInitializer.InitDone()
+}
```

Comment:
`NodeSync` may be called twice if the kvstore is enabled (once when the informer syncs, the other when the kvstore syncs). However, `InitDone` is implemented on top of a wait group, which panics if it turns negative.

Reply:
Additionally, this logic is not accounting for the clustermesh case. You can retrieve whether clustermesh is synchronized using `(*ClusterMesh).NodesSynced()`.

Reply:
Yeah, the fact that we can flip back and forth from kvstore to crd is not playing nice with the initializers logic.

> I think we should either make sure that InitDone is called exactly once, or make it idempotent (i.e., a no-op after the first execution).

I think the first requires an ad-hoc solution for the kvstore/crd case, it is probably better to make the `InitDone` idempotent, so to keep having one "signal" for each "initializer" taken from the manager.

I'll try with the `StoppableWaitGroup`, thanks for the heads up!